tmk_core/common/mousekey.c

```diff
@@ -114,7 +114,9 @@ void mousekey_task(void) {
   /* diagonal move [1/sqrt(2)] */
   if (mouse_report.x && mouse_report.y) {
     mouse_report.x = times_inv_sqrt2(mouse_report.x);
+    mouse_report.x = mouse_report.x == 0 ? 1 : mouse_report.x;
```

Comment:
This is harder to read than it needs to be. I suggest replacing it with:
```suggestion
    if (mouse_report.x == 0) { mouse_report.x = 1; }
```